numba/np/arrayobj.py

```diff
@@ -3399,8 +3400,13 @@ def _empty_nd_impl(context, builder, arrtype, shapes):
         )
 
     align = context.get_preferred_array_alignment(arrtype.dtype)
-    meminfo = context.nrt.meminfo_alloc_aligned(builder, size=allocsize,
-                                                align=align)
+    def alloc_unsupported(context, builder, size, align):
+        return context.nrt.meminfo_alloc_aligned(builder, size, align)
+
+    # See if the type has a special allocator, if not use the default
+    # alloc_unsuppported allocator above.
+    allocator_impl = _allocators.lookup(arrtype.__class__, alloc_unsupported)
+    meminfo = allocator_impl(context, builder, size=allocsize, align=align)
```

Comment:
I wonder if we can refactor the core compiler to better support custom allocators

Reply:
I am the least sure on this part.  This spot is the right spot to intercept allocation for a type because the allocation type doesn't propagate further into the allocators so you can't really intercept it where you'd like to only use a different allocator.  Memory management and allocation are different and right now we're forced to overload mem-mgmt when we'd only like to do the allocator.  I'm definitely open to all your suggestions if you want to do a more significant rearchitecting.